mw-webapp/src/logic/wayPage/WayPage.tsx

```diff
@@ -1,24 +1,127 @@
 import {GoalPreview} from "src/model/businessModelPreview/GoalPreview";
+import {UserPreview} from "src/model/businessModelPreview/UserPreview";
 import {WayPreview} from "src/model/businessModelPreview/WayPreview";
 import {pages} from "src/router/pages";
 import styles from "src/logic/wayPage/WayPage.module.scss";
 
 /**
  * Change description of Way
  */
-const updateGoalWay = (wayPreview: WayPreview, description: string) => {
-  const newGoal = new GoalPreview({...wayPreview.goal, description});
-  GoalPreviewDAL.updateGoalPreview(newGoal);
+const changeGoalDescription = (wayPreview: WayPreview, description: string) => {
+  const newGoalPreview = new GoalPreview({...wayPreview.goal, description});
+  GoalPreviewDAL.updateGoalPreview(newGoalPreview);
+};
+
+/**
+ * Change name of Way
+ */
+const changeWayName = (wayPreview: WayPreview, name: string) => {
+  const newWayPreview = new WayPreview({...wayPreview, name});
+  WayPreviewDAL.updateWayPreview(newWayPreview);
+};
+
+/**
+ * Add mentor to Way
+ */
+const addMentorToWay = (
+  wayPreview: WayPreview,
+  setWay: React.Dispatch<React.SetStateAction<WayPreview | null>>,
+  userPreview: UserPreview,
+) => {
+  const mentoringWays = userPreview.mentoringWays.concat(userPreview.uuid);
+  const newUserPreview = new UserPreview({...userPreview, mentoringWays});
+
+  UserPreviewDAL.updateUserPreview(newUserPreview);
+
+  const mentors = wayPreview.mentors.concat(newUserPreview);
+  const newWayPreview = new WayPreview({...wayPreview, mentors});
+
+  WayPreviewDAL.updateWayPreview(newWayPreview);
+
+  setWay(newWayPreview);
+};
+
+/**
+ * Add user to Way's mentor requests
+ */
+const addUserToMentorRequests = (
+  wayPreview: WayPreview,
+  setWay: React.Dispatch<React.SetStateAction<WayPreview | null>>,
+  userPreview: UserPreview) => {
+
+  const mentorRequests = wayPreview.mentorRequests.concat(userPreview);
+
+  const newWayPreview = new WayPreview({...wayPreview, mentorRequests});
+
+  WayPreviewDAL.updateWayPreview(newWayPreview);
+  setWay(newWayPreview);
+};
+
+/**
+ * Remove user from Way's mentor requests
+ */
+const removeUserFromMentorRequests = (
+  wayPreview: WayPreview,
+  setWay: React.Dispatch<React.SetStateAction<WayPreview | null>>,
+  userPreview: UserPreview) => {
+
+  const mentorRequests = wayPreview.mentorRequests.filter((item) => item !== userPreview);
+
+  const newWayPreview = new WayPreview({...wayPreview, mentorRequests});
+
+  WayPreviewDAL.updateWayPreview(newWayPreview);
+  setWay(newWayPreview);
+};
+
+/**
+ * Render all Way's mentors
+ */
+const renderMentors = (wayPreview: WayPreview) => {
+  return wayPreview.mentors.map((item) => (
+    <Link
+      key={item.uuid}
+      value={item.name}
+      path={pages.user.getPath({uuid: item.uuid})}
+      className={styles.mentors}
+    />
+  ));
+};
+
+/**
+ * Render Way's mentor requests
+ */
+const renderMentorRequests = (wayPreview: WayPreview, setWay: React.Dispatch<React.SetStateAction<WayPreview | null>>) => {
```

Comment:
Does it possible to set the way as null?

Reply:
Yes, because it's a state